Monika-After-Story/MonikaModDev
# New 922

Would you look at that, it's a small event PR! Unheard of!

## Testing:
- Verify new `gotcha` note works. It should appear after you confirm the party w/ chibika. It should be deleted after the party is over.
- Verify Black Dress intro works for subsequent years with appropriate dialogue.
- Verify cakes work properly, they now take the current filter. Verify they still get lit up (This should be tested for normal bday and player bday)
- Flag a background as `EXP_TYPE_OUTDOOR` (via its exprops). Bday deco should only be balloons and cake. When switching to an indoor background, the deco should return to banners, cake, and balloons.
